src/bvh/distance_traverse.rs

```diff
@@ -46,133 +63,75 @@ where
                 child_l_index,
                 ref child_l_aabb,
                 child_r_index,
                 ref child_r_aabb,
                 ..
             } => {
-                let left_dist = self
-                    .ray
-                    .intersection_slice_for_aabb(child_l_aabb)
-                    .map(|(d, _)| d);
-                let right_dist = self
-                    .ray
-                    .intersection_slice_for_aabb(child_r_aabb)
-                    .map(|(d, _)| d);
-
-                match (left_dist, right_dist) {
-                    (None, None) => {
-                        // no intersections with any children
-                        self.has_node = false;
-                        (current_node_index, RestChild::None)
-                    }
-                    (Some(_), None) => {
-                        // has intersection only with left child
-                        self.has_node = true;
-                        self.node_index = child_l_index;
-                        let rest_child = RestChild::None;
-                        (current_node_index, rest_child)
-                    }
-                    (None, Some(_)) => {
-                        // has intersection only with right child
-                        self.has_node = true;
-                        self.node_index = child_r_index;
-                        let rest_child = RestChild::None;
-                        (current_node_index, rest_child)
-                    }
-                    (Some(left_dist), Some(right_dist)) => {
-                        if (left_dist > right_dist) ^ !ASCENDING {
-                            // right is higher priority than left
-                            self.has_node = true;
-                            self.node_index = child_r_index;
-                            let rest_child = RestChild::Left;
-                            (current_node_index, rest_child)
-                        } else {
-                            // left is higher priority than right
-                            self.has_node = true;
-                            self.node_index = child_l_index;
-                            let rest_child = RestChild::Right;
-                            (current_node_index, rest_child)
-                        }
-                    }
-                }
-            }
-            BvhNode::Leaf { .. } => {
-                self.has_node = false;
-                (current_node_index, RestChild::None)
+                let left_dist_opt = self.ray.intersection_slice_for_aabb(child_l_aabb);
+                let right_dist_opt = self.ray.intersection_slice_for_aabb(child_r_aabb);
+
+                if let Some((left_dist_min, left_dist_max)) = left_dist_opt {
+                    // left child has intersection with ray
+                    let left_dist = if ASCENDING {
+                        left_dist_min
+                    } else {
+                        left_dist_max
+                    };
```

Comment:
I think this deserves a comment *somewhere*. Perhaps the public documentation should mention that farthest-traversal considers the distance to the exit point of the ray. Perhaps that should also be mentioned here.